integration/test/metric_value_benchmark/metrics_value_benchmark_test.go

```diff
@@ -10,83 +10,53 @@ import (
+const agentConfigDirectory = "agent_configs"
+const agentConfigFileName = "/base_config.json"
 
 const namespace = "MetricValueBenchmarkTest"
 const instanceId = "InstanceId"
 
 const minimumAgentRuntime = 3 * time.Minute
 
-func TestCPUValue(t *testing.T) {
-	log.Printf("testing cpu value...")
-
-	resourcePath := "agent_configs"
-
-	log.Printf("resource file location %s", resourcePath)
-
-	t.Run(fmt.Sprintf("resource file location %s ", resourcePath), func(t *testing.T) {
-		test.CopyFile(resourcePath+configJSON, configOutputPath)
-		err := test.StartAgent(configOutputPath, false)
-
-		if err != nil {
-			t.Fatalf("Agent could not start")
-		}
-
-		time.Sleep(minimumAgentRuntime)
-		log.Printf("Agent has been running for : %s", minimumAgentRuntime.String())
-		test.StopAgent()
-
-		testResult := validateCpuMetrics()
-		testSuiteStatus := getTestSuiteStatus(testResult)
-		printTestResult(testSuiteStatus, testResult)
-
-		if testSuiteStatus == status.FAILED {
-			t.Fatalf("Cpu test failed to validate that every metric value is greater than zero")
-		}
-	})
-
-	// TODO: Get CPU value > 0
-	// TODO: Range test with >0 and <100
-	// TODO: Range test: which metric to get? api reference check. should I get average or test every single datapoint for 10 minutes? (and if 90%> of them are in range, we are good)
+type MetricBenchmarkTestSuite struct {
+	suite.Suite
+	result status.TestSuiteResult
 }
 
-var metricsToFetch = []string{
-	"cpu_time_active", "cpu_time_guest", "cpu_time_guest_nice", "cpu_time_idle", "cpu_time_iowait", "cpu_time_irq",
-	"cpu_time_nice", "cpu_time_softirq", "cpu_time_steal", "cpu_time_system", "cpu_time_user",
-	"cpu_usage_active", "cpu_usage_quest", "cpu_usage_quest_nice", "cpu_usage_idle", "cpu_usage_iowait",
-	"cpu_usage_irq", "cpu_usage_nice", "cpu_usage_softirq", "cpu_usage_steal", "cpu_usage_system", "cpu_usage_user"}
-
-func validateCpuMetrics() map[string]status.TestStatus {
-	validationResult := map[string]status.TestStatus{}
-	for _, metricName := range metricsToFetch {
-		validationResult[metricName] = status.FAILED
+func (suite *MetricBenchmarkTestSuite) SetupSuite() {
+	fmt.Println(">>>> Starting MetricBenchmarkTestSuite")
+}
 
-		fetcher, err := metric.GetMetricFetcher(metricName)
-		if err != nil {
-			continue
-		}
+func (suite *MetricBenchmarkTestSuite) TearDownSuite() {
+	suite.result.Print()
+	fmt.Println(">>>> Finished MetricBenchmarkTestSuite")
+}
 
-		values, err := fetcher.Fetch(namespace, metricName, metric.AVERAGE)
-		if err != nil {
-			continue
-		}
+func TestMetricValueBenchmarkSuite(t *testing.T) {
+	suite.Run(t, new(MetricBenchmarkTestSuite))
+}
 
-		if !isAllValuesGreaterThanZero(metricName, values) {
-			continue
-		}
+func (suite *MetricBenchmarkTestSuite) RunAgent(agentConfigFileName string, runningDuration time.Duration) {
+	agentConfigPath := agentConfigDirectory + agentConfigFileName
+	log.Printf("Starting agent using agent config file %s", agentConfigPath)
```

Comment:
wrt the open question about printing, `testing.T` has a `Log/Logf` function that can be used. Not sure if that helps at all

Reply:
Might try and see! Thanks. I'm not yet aware of the difference between all these different ways of logging